### ISSUE TITLE
Traits with lifetimes fail to compile

### ISSUE THREAD

**Issue by WilliamVenner** [jmg-duarte/sealed-rs#12]
```rust
#[sealed::sealed]
pub trait Testing<'a> {}

error[E0106]: missing lifetime specifier
  |
1 | #[sealed::sealed]
  | ^ expected named lifetime parameter
  |
  = note: this error originates in the attribute macro `sealed::sealed` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider using the `'a` lifetime
  |
1 | #<'a>[sealed::sealed]
  | ~~~~~

For more information about this error, try `rustc --explain E0106`.
```

**Comment by jmg-duarte** [comment]
Could you provide the `cargo expand` output?

**Comment by jmg-duarte** [comment]
@WilliamVenner I have a fix for your issue in particular but I'd like a better example to solidify the implementation.
If you could submit one, it would be great!

**Comment by MingweiSamuel** [comment]
The issue is that here only `type_params` are forwarded, not lifetimes or const generic params:
https://github.com/jmg-duarte/sealed-rs/blob/aea9c3db945db1ff535ccd1c3a49f7663b0c1a8e/src/lib.rs#L157-L159
The fix (#13) is to use `let (_, ty_generics, _) = trait_generics.split_for_impl();` and `#seal::Sealed #ty_generics` which includes all three types of params together.
